egui/src/data/output.rs

```diff
@@ -317,9 +351,24 @@ impl WidgetInfo {
     }
 
     #[allow(clippy::needless_pass_by_value)]
-    pub fn text_edit(edit_text: impl ToString) -> Self {
+    pub fn text_edit(text_value: impl ToString, prev_text_value: impl ToString) -> Self {
+        Self {
+            text_value: Some(text_value.to_string()),
+            prev_text_value: Some(prev_text_value.to_string()),
+            ..Self::new(WidgetType::TextEdit)
+        }
+    }
+
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn text_selection_changed(
+        primary_cursor: usize,
+        secondary_cursor: usize,
```

Comment:
```suggestion
        char_range: std::Ops::Range<usize>,
```

Data that belong together should be clumped together :)